### ISSUE TITLE
Arbitrary number properties are allowed

### ISSUE THREAD

**Issue by vkurchatkin** [facebook/flow#1512]
This type-checks:

``` js
type O = {};

function stuff(o: O) {
  const b = o[0].ff.ffa
}
```

`o[0]` behaves as it it was declared as `any`

**Comment by vkurchatkin** [comment]
This is still an issue.

**Comment by JohnyDays** [comment]
I think, if you want this to type-check you have to seal the object. E.g `type O = { [key: string]: void };`

**Comment by vkurchatkin** [comment]
Nope, this is definitely a bug, for example changing `[0]` to `['0']` fixes the problem

**Comment by gkz** [comment]
Fixed in 80ca16f77ba81a3475dd1f7a9f4d9d9e3d71bca8
